pkg/apis/application/v1alpha1/types.go

```diff
@@ -2467,6 +2487,13 @@ func (c *Cluster) RawRestConfig() *rest.Config {
 	if err != nil {
 		panic(fmt.Sprintf("Unable to create K8s REST config: %v", err))
 	}
+	if c.Config.ProxyUrl != "" {
+		u, err := c.ParseProxyUrl(c.Config.ProxyUrl)
+		if err != nil {
+			panic(fmt.Sprintf("Unable to use proxy: %v", err))
```

Comment:
I think we should change the method signature and propagate this up instead of panicking.